src/databricks/sqlalchemy/test/_unsupported.py

```diff
@@ -0,0 +1,430 @@
+    SYMBOL_CHARSET = "symbols expected by test"
+    TEMP_TBL = "temporary tables"
+    TRANSACTIONS = "transactions"
+    UNIQUE = "UNIQUE constraints"
+
+
+def render_skip_reason(rsn: SkipReason, setup_error=False, extra=False) -> str:
+    prefix = "[BADSETUP]" if setup_error else ""
+    postfix = " More detail in _unsupported.py" if extra else ""
+    return f"[UNSUPPORTED]{prefix}[{rsn.name}]: Databricks does not support {rsn.value}.{postfix}"
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.ENFORCE_KEYS))
+class ExceptionTest(ExceptionTest):
+    """Per Databricks documentation, primary and foreign key constraints are informational only
+    and are not enforced.
+
+    https://docs.databricks.com/api/workspace/tableconstraints
+    """
+
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.IDENTIFIER_LENGTH))
+class LongNameBlowoutTest(LongNameBlowoutTest):
+    """These tests all include assertions that the tested name > 255 characters"""
+
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.SEQUENCES))
+class HasSequenceTest(HasSequenceTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.SEQUENCES))
+class HasSequenceTestEmpty(HasSequenceTestEmpty):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.INDEXES))
+class HasIndexTest(HasIndexTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.SYMBOL_CHARSET))
+class UnicodeSchemaTest(UnicodeSchemaTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.CURSORS))
+class ServerSideCursorsTest(ServerSideCursorsTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.SYMBOL_CHARSET))
+class PercentSchemaNamesTest(PercentSchemaNamesTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.TRANSACTIONS))
+class IsolationLevelTest(IsolationLevelTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.TRANSACTIONS))
+class AutocommitIsolationTest(AutocommitIsolationTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.RETURNING))
+class ReturningTest(ReturningTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.SEQUENCES))
+class SequenceTest(SequenceTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.SEQUENCES))
+class SequenceCompilerTest(SequenceCompilerTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.GENERATED_COLUMNS))
+class ComputedColumnTest(ComputedColumnTest):
+    pass
+
+
+@pytest.mark.reviewed
+@pytest.mark.skip(reason=render_skip_reason(SkipReason.GENERATED_COLUMNS))
```

Comment:
Is this feature supported in the DBR / do we plan to support it?  I have a ticket on dbt-databricks about it.

Reply:
Oh nice catch. After I wrote this skip marker a public preview actually went live for this feature.  https://docs.databricks.com/en/delta/generated-columns.html

I can move this from _unsupported to _future. Implementation will be tricky since computed columns can use any Spark function, which SQLAlchemy is unaware of. Will be interesting to see how we can implement this.

Reply:
I've moved the generated columns tests from unsupported to future here.

[fc3aa37](https://github.com/databricks/databricks-sql-python/pull/256/commits/fc3aa374507ae7dc18b651dce666d42339c4bc0e)